packages/kbn-monaco/src/esql/lib/esql_theme.test.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ESQLErrorListener, getLexer as _getLexer } from '@kbn/esql-ast';
+import { ESQL_TOKEN_POSTFIX } from './constants';
+import { buildESQlTheme } from './esql_theme';
+import { CharStreams } from 'antlr4';
+
+describe('ESQL Theme', () => {
+  it('should not have multiple rules for a single token', () => {
+    const theme = buildESQlTheme();
+
+    const seen = new Set<string>();
+    const duplicates: string[] = [];
+    for (const rule of theme.rules) {
+      if (seen.has(rule.token)) {
+        duplicates.push(rule.token);
+      }
+      seen.add(rule.token);
+    }
+
+    expect(duplicates).toEqual([]);
+  });
+
+  const getLexer = () => {
+    const errorListener = new ESQLErrorListener();
+    const inputStream = CharStreams.fromString('FROM foo');
+    return _getLexer(inputStream, errorListener);
+  };
+
+  const lexer = getLexer();
+  const lexicalNames = lexer.symbolicNames
+    .filter((name) => typeof name === 'string')
+    .map((name) => name!.toLowerCase());
+
+  it('every rule should apply to a valid lexical name', () => {
+    const theme = buildESQlTheme();
+
+    // These names aren't from the lexer... they are added on our side
+    // see packages/kbn-monaco/src/esql/lib/esql_token_helpers.ts
+    const syntheticNames = ['functions', 'nulls_order', 'timespan_literal'];
+
+    for (const rule of theme.rules) {
+      expect([...lexicalNames, ...syntheticNames]).toContain(
+        rule.token.replace(ESQL_TOKEN_POSTFIX, '').toLowerCase()
+      );
+    }
+  });
+
+  it('every valid lexical name should have a corresponding rule', () => {
```

Comment:
Do we like this test? It seemed like a good idea as an extra check that we aren't missing some lexical token that gets added. But, we can remove if it seems like a pain.

Reply:
Seems painful to me 🙈